ui/app/controllers/vault/cluster/clients/counts.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Controller from '@ember/controller';
+import { action, set } from '@ember/object';
+
+import type { ClientsCountsRouteParams } from 'vault/routes/vault/cluster/clients/counts';
+
+const queryParamKeys = ['start_time', 'end_time', 'ns', 'authMount'];
+export default class ClientsCountsController extends Controller {
+  queryParams = queryParamKeys;
+
+  start_time: string | number | undefined = undefined;
+  end_time: string | number | undefined = undefined;
+  ns: string | undefined = undefined;
+  authMount: string | undefined = undefined;
+
+  // using router.transitionTo to update the query params results in the model hook firing each time
+  // this happens when the queryParams object is not added to the route or refreshModel is explicitly set to false
+  // updating the bound properties does however respect the refreshModel settings and functions expectedly
+  @action
+  updateQueryParams(updatedParams: ClientsCountsRouteParams) {
+    if (!updatedParams) {
+      this.queryParams.forEach((key) => (this[key as keyof ClientsCountsRouteParams] = undefined));
+    } else {
+      Object.keys(updatedParams).forEach((key) => {
+        if (queryParamKeys.includes(key)) {
+          const value = updatedParams[key as keyof ClientsCountsRouteParams];
+          set(this, key as keyof ClientsCountsRouteParams, value as keyof ClientsCountsRouteParams);
+        }
+      });
+    }
+  }
```

Comment:
This was something interesting where totally removing the `queryParams` block from the route file and not having `refreshModel` set was still resulting in the model hook firing when doing a `transitionTo` and updating the query params.

There have been talks to do the actual filtering of the activity in the model hook in which case this can be removed but for now updating `ns` and `authMount` query params does not trigger a full transition.